client/my-sites/stats/hooks/use-feature-gating.ts

```diff
@@ -0,0 +1,42 @@
+import { isEnabled } from '@automattic/calypso-config';
+import { FEATURE_STATS_PAID } from '@automattic/calypso-products';
+import { useSelector } from 'calypso/state';
+import siteHasFeature from 'calypso/state/selectors/site-has-feature';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+const trafficPaidStats = [
+	'statsSearchTerms',
+	'statsClicks',
+	'statsReferrers',
+	'statsCountryViews',
+];
+
+const featureFlags = [ 'stats/date-control' ];
+
+/*
+ * Utility function intended to be used with useSelector or redux connect mapStateToProps.
+ * For example in mapStateToProps:
+ * const canAccessFeature = checkFeatureGating( state, siteId, 'statsSearchTerms' );
+ */
+export const checkFeatureGating = ( state: object, siteId: number | null, featureType: string ) => {
```

Comment:
Are these statTypes a ts type anywhere we could use?

Reply:
I dug around and found [this](https://github.com/Automattic/wp-calypso/blob/trunk/packages/wpcom.js/src/lib/runtime/site.get.js#L2). I'll add a file where we can make them into variables. Currently, `statType`s in calypso are hardcoded.

Reply:
We could also do ⬆️ in a separate PR so that you can use the functions with your implementation.

Reply:
Sure lets do that in a follow up. Don't worry too much about rushing this though, it's not really blocking and I think it's worth getting these things right the first time: https://github.com/Automattic/wp-calypso/pull/85041#issuecomment-1851059191